CMakeLists.txt

```diff
@@ -1,139 +1,199 @@
-# Is this a build as part of Trilinos?
 
+# We want to determine if options are given with the wrong case
+# In order to detect which arguments are given to compare against
+# the list of valid arguments, at the beginning here we need to
+# form a list of all the given variables. If it begins with any 
+# case of KoKkOS, we add it to the list.
+GET_CMAKE_PROPERTY(_variableNames VARIABLES)
+SET(KOKKOS_GIVEN_VARIABLES)
+FOREACH (var ${_variableNames})
+  STRING(TOUPPER ${var} UC_VAR)
+  STRING(FIND ${UC_VAR} KOKKOS IDX)
+  IF (${IDX} EQUAL 0)
+    LIST(APPEND KOKKOS_GIVEN_VARIABLES ${var})
+  ENDIF()
+ENDFOREACH()
+
+# Is this a build as part of Trilinos?
 IF(COMMAND TRIBITS_PACKAGE_DECL)
   SET(KOKKOS_HAS_TRILINOS ON CACHE BOOL "")
```

Comment:
Is this an option the average user should see? If so, it would be really useful to have a docstring. Also, it is a bit weird that this is the only option that is spelled in all caps.

Reply:
This is not a user option. This should probably just be a regular variable that doesn't show up in cache (this is a carry-over from the original CMake).